src/model.js

```diff
@@ -2613,6 +2613,11 @@ class Model {
     const updateValues = Utils.mapValueFieldNames(updatedDataValues, options.fields, this);
     const now = Utils.now(this.sequelize.options.dialect);
 
+    options.conflictFields = options.conflictFields
+      && options.conflictFields.map(
+        attr => (this.rawAttributes[attr] && this.rawAttributes[attr].field) || attr,
+      );
+
```

Comment:
just a small code style change

```suggestion
    options.conflictFields = options.conflictFields?.map(
      attr => (this.rawAttributes[attr]?.field) || attr,
    );

```

This piece of code is used a lot in sequelize. We could add a `Model.getColumnName(attributeName: string): string` util which does this